File: es2/es2.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	input, err := os.Open("input.txt")
	if err != nil {
		panic(err)
	}
	defer input.Close()

	twos := 0
	threes := 0
	scanner := bufio.NewScanner(input)
	for scanner.Scan() {
		line := scanner.Text()
		letters := ['z' - 'a' + 1]byte{0}
		for _, byte := range line {
			if byte <= 'z' && byte >= 'a' {
				letters[byte-'a']++
			}
		}

		found2 := false
		found3 := false
		for _, count := range letters {
			if count == 2 {
				found2 = true
			} else if count == 3 {
				found3 = true
			}
		}
		if found2 {
			twos++
		}
		if found3 {
			threes++
		}
	}

	fmt.Println("Checksum:", twos*threes)
}

File: es2/es2.py
#!/usr/bin/env python3

two = 0
three = 0
with open("input.txt", "r") as f:
    for line in f:
        letters = {}
        for char in line:
            if char in letters:
                letters[char] += 1
            else:
                letters[char] = 1

        found2 = False
        found3 = False
        for letter, count in letters.items():
            if count == 2:
                found2 = True
            elif count == 3:
                found3 = True
        if found2:
            two += 1
        if found3:
            three += 1

    print("Checksum:", two*three)
